servicediscovery/mdns/mdns.go

```diff
@@ -0,0 +1,66 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package mdns
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/dapr/components-contrib/servicediscovery"
+	"github.com/grandcat/zeroconf"
+)
+
+func NewmDNSResolver() *Resolver {
+	return &Resolver{}
+}
+
+type Resolver struct {
+}
+
+func (z *Resolver) ResolveID(req *servicediscovery.ResolveRequest) (string, error) {
+	port, err := LookupPortMDNS(req.ID)
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("localhost:%v", port), nil
+}
+
+// LookupPortMDNS uses mdns to find the port of a given service entry on a local network
+func LookupPortMDNS(id string) (int, error) {
+	resolver, err := zeroconf.NewResolver(nil)
```

Comment:
Do we need to create resolver instance every time? can't we cache it?

Reply:
It should be possible to cache this.

Reply:
Put inside a once.Do